src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs

```diff
@@ -1846,36 +1846,64 @@ private static void CheckStringSplitOptions(StringSplitOptions options)
 
         // Returns a substring of this string.
         //
-        public string Substring(int startIndex) => Substring(startIndex, Length - startIndex);
-
-        public string Substring(int startIndex, int length)
+        public string Substring(int startIndex)
         {
-            if (startIndex < 0)
+            if (startIndex == 0)
             {
-                throw new ArgumentOutOfRangeException(nameof(startIndex), SR.ArgumentOutOfRange_StartIndex);
+                return this;
             }
 
-            if (startIndex > Length)
-            {
-                throw new ArgumentOutOfRangeException(nameof(startIndex), SR.ArgumentOutOfRange_StartIndexLargerThanLength);
-            }
+            int length = Length;
+            // Copy Span.Slice checks for perf
+            if ((uint)startIndex > (uint)length)
+                ThrowHelper.ThrowArgumentOutOfRangeException();
 
-            if (length < 0)
-            {
-                throw new ArgumentOutOfRangeException(nameof(length), SR.ArgumentOutOfRange_NegativeLength);
-            }
+            return InternalSubString(startIndex, length - startIndex);
+        }
 
-            if (startIndex > Length - length)
-            {
-                throw new ArgumentOutOfRangeException(nameof(length), SR.ArgumentOutOfRange_IndexLength);
-            }
+        public string Substring(int startIndex, int length)
+        {
+            // Copy Span.Slice checks for perf
+            int _length = Length;
```

Comment:
Nit: We don't tend to use underscores for local variables. Consider using "thisLength" or "capturedLength" or similar.

Reply:
Sorry, I know this was just quick changes, quick copy from span, will try to clean before a final review, just wanted to be sure I was on the right track before spending more time on this. :)